src/opendr/perception/pose_estimation/hr_pose_estimation/high_resolution_learner.py

```diff
@@ -424,6 +619,348 @@ def eval(self, dataset, silent=False, verbose=True, use_subset=True, subset_size
+
+                            total_keypoints_num = 0
+                            all_keypoints_by_type = []
+                            for kpt_idx in range(18):
+                                total_keypoints_num += extract_keypoints(avg_heatmaps[:, :, kpt_idx], all_keypoints_by_type,
+                                                                         total_keypoints_num)
+
+                            pose_entries, all_keypoints = group_keypoints(all_keypoints_by_type, avg_pafs)
+
+                            for kpt_id in range(all_keypoints.shape[0]):
+                                all_keypoints[kpt_id, 0] = ((all_keypoints[kpt_id, 0] *
+                                                             self.stride / upsample_ratio - pad[1]) / scale)
+                                all_keypoints[kpt_id, 1] = ((all_keypoints[kpt_id, 1] *
+                                                             self.stride / upsample_ratio - pad[0]) / scale)
+
+                            for i in range(all_keypoints.shape[0]):
+                                for j in range(all_keypoints.shape[1]):
+                                    if j == 0:
+                                        all_keypoints[i][j] = round((all_keypoints[i][j] + xmin))
+                                    if j == 1:
+                                        all_keypoints[i][j] = round((all_keypoints[i][j] + ymin))
+
+                            current_poses = []
+                            for n in range(len(pose_entries)):
+                                if len(pose_entries[n]) == 0:
+                                    continue
+                                pose_keypoints = np.ones((num_keypoints, 2), dtype=np.int32) * -1
+                                for kpt_id in range(num_keypoints):
+                                    if pose_entries[n][kpt_id] != -1.0:  # keypoint was found
+                                        pose_keypoints[kpt_id, 0] = int(all_keypoints[int(pose_entries[n][kpt_id]), 0])
+                                        pose_keypoints[kpt_id, 1] = int(all_keypoints[int(pose_entries[n][kpt_id]), 1])
+                                pose = Pose(pose_keypoints, pose_entries[n][18])
+                                current_poses.append(pose)
+
+                            coco_keypoints, scores = convert_to_coco_format(pose_entries, all_keypoints)
+
+                            image_id = int(file_name[0:file_name.rfind('.')])
+
+                            for idx in range(len(coco_keypoints)):
+                                coco_result.append({
+                                    'image_id': image_id,
+                                    'category_id': 1,  # person
+                                    'keypoints': coco_keypoints[idx],
+                                    'score': scores[idx]
+                                })
+
+                        if self.visualize:
+                            for keypoints in coco_keypoints:
```

Comment:
`coco_keypoints` seems to be assigned within `if crop_size > 0` above. Is there a case where `self.visualize == True` and `crop_size <= 0` where this might throw an error? If not, ignore this comment